### PR TITLE
Restaurant Search

### PR DESCRIPTION
**Pull request by M7Mansour**
build: created a route, and controller for searching a specific restaurant

### CODE REVIEW

File: server/src/router.js [hunk 1]
```diff
@@ -19,6 +20,7 @@ router.post('/promo', decodeToken, createPromoCode);
 router.get('/meals/category/:id', getMealsByCategory);
 router.get('/meals/restaurant/:id', getMealsByRestaurant);
 router.get('/meals/:mealid', getMealById);
+router.get('/restaurants/search/:name', searchRestaurant);
```
Comment (Amoodaa) [inline server/src/router.js:23]: how about GET `/restaurants?search=<query>

**Review by Amoodaa — APPROVED** [review]

File: server/src/router.js (hunk 1 above)
Comment (M7Mansour) [reply server/src/router.js:23]: wooooow
greate idea 
I have to edit all the code now :)

Comment (Amoodaa) [reply server/src/router.js:23]: the route will be `GET /restaurants`

Comment (Amoodaa) [reply server/src/router.js:23]: querystrings can be accessed from req.query

Comment (M7Mansour) [reply server/src/router.js:23]: yea yea I understand

File: server/src/controllers/restaurants.js [hunk 2]
```diff
@@ -12,10 +12,21 @@ const getRestaurants = async (req, res, next) => {
       );
 
     const [count, data] = await Promise.all([
-      Restaurant.count(),
+      Restaurant.count({
+        where: {
+          restaurantName: {
```
Comment (Amoodaa) [inline server/src/controllers/restaurants.js:17]: whats restaurantName

Comment (Amoodaa) [reply server/src/controllers/restaurants.js:17]: why is it not just `name`

Comment (M7Mansour) [reply server/src/controllers/restaurants.js:17]: this is its name in the schema

**Review by Amoodaa — APPROVED** [review]

**Review by NourAdawi — APPROVED** [review]

**Pull request merged**